src/transformState.js

```diff
@@ -2,10 +2,40 @@
 
 /**
  * @param {Object} state
+ * @typeof {Object} Action
+ * @property {string} type
+ * @property {Object} extraData
+ * @property {string[]} keysToRemove
  * @param {Object[]} actions
  */
+
+const ADD_PROPERTIES = 'addProperties';
+const REMOVE_PROPERTIES = 'removeProperties';
+const CLEAR = 'clear';
+
 function transformState(state, actions) {
-  // write code here
+  actions.forEach((action) => {
+    switch (action.type) {
+      case ADD_PROPERTIES:
+        Object.assign(state, action.extraData);
+        break;
+      case REMOVE_PROPERTIES:
+        action.keysToRemove.forEach((key) => {
+          if (state.hasOwnProperty(key)) {
+            delete state[key];
+          }
+        });
+        break;
+      case CLEAR:
+        Object.keys(state).forEach((key) => {
+          delete state[key];
+        });
+        break;
+      default:
```

Comment:
default should throw new Error when reached

Reply:
Yeah but lint don't letting me use console.log or console.warn here

Reply:
nevermind I will update this in a sec